core/validation/utils.py

```diff
@@ -12,3 +16,94 @@ def remove_duplicate_indexes(df: pd.DataFrame):
     :return: the DataFrame without duplicate indexes
     """
     return df[~df.index.duplicated(keep="first")]
+
+
+def get_start_date_column_for_failures(failed_rows: pd.DataFrame) -> list[tuple] | None:
+    """
+    Retrieves the 'Start_Date' column values for the specified indexes.
+
+    :param failed_rows: A pandas DataFrame containing the data of rows that have failed validation.
+    :return: A list of tuples, where each tuple contains an index and the corresponding 'Start_Date' value.
+    """
+    start_dates_series = failed_rows["Start_Date"] if "Start_Date" in failed_rows.columns else None
+    if start_dates_series is not None:
+        test = [(index, str(start_date)) for index, start_date in start_dates_series.iteritems()]
+        return test
+
+
+def get_cell_indexes_for_outcomes(failed_rows: list[tuple]) -> str:
+    """
+    Constructs cell indexes for outcomes based on the provided list of failed rows.
+
+    :param failed_rows: A list of tuples, each containing an index and the corresponding 'Start_Date' value.
+    :return: A string containing the constructed cell indexes for outcomes.
+    """
+    cell_indexes = []
+
+    for index, start_date in failed_rows:
+        # footfall outcomes starts from row 60
+        if index >= 60:
+            financial_year = get_uk_financial_year_start(start_date)
+            # row for 'Amount' column is end number of start year of financial year * 5 + 'Footfall Indicator' index
+            row_index_gap = int(financial_year[-1]) * 5
+            index = str(int(index) + row_index_gap)
+            cell_index = footfall_outcome_amount_to_col(start_date).format(i=index)
+            cell_indexes.append(cell_index)
+        else:
+            index = str(index)
+            cell_index = non_footfall_outcome_amount_to_col(start_date).format(i=index)
+            cell_indexes.append(cell_index)
+
+    return ", ".join(cell_indexes)
+
+
+def get_uk_financial_year_start(start_date: str) -> str:
+    """
+    Gets the start year of the UK financial year based on the provided start date.
+
+    :param start_date: A string representing the start date in the format '%Y-%m-%d %H:%M:%S'.
+    :return: A string representing the start year of the UK financial year.
+    """
+    start_date = datetime.strptime(start_date, "%Y-%m-%d %H:%M:%S")
+
+    financial_year_start = datetime(start_date.year, 4, 1)
+    if start_date < financial_year_start:
+        financial_year_start = datetime(start_date.year - 1, 4, 1)
+
+    return f"{str(financial_year_start.year)}"
+
+
+def get_month(start_date: str) -> str:
+    """
+    Gets the month name based on the provided start date.
+
+    :param start_date: A string representing the start date in the format '%Y-%m-%d %H:%M:%S'.
+    :return: A string representing the month name.
+    """
+    start_date = datetime.strptime(start_date, "%Y-%m-%d %H:%M:%S")
+
+    return start_date.strftime("%B")
```

Comment:
This can just be `start_date.month`

Reply:
I would argue that `str(start_date.month)` is clearer than `start_date.strftime("%B")`

It's not immediately obvious what `%B` means

Reply:
```str(start_date.month)``` returns a string of '5' if the month is May, and I think the string representation of the month's name is clearer.

We can do ```month_name = calendar.month_name[start_date.month]```, which is perhaps a little clearer as to the intention of the code.

Reply:
Ah I see what you mean now sorry. That looks good 👍